### ISSUE TITLE
Model Brush Limit (triggers, func_buttons, ect) effectively 999, not 1024 or 4096 (All versions)

### ISSUE THREAD

**Issue by Armystuntman** [Facepunch/garrysmod-issues#4602]
### Details
Hammer does not compile any brush entities from 1000 onwards. While Hammer follows these limits, there are no errors given about 1000+ brush entities but they are not seen in-game. Only 999 can possibly show up in-game. Tested on all branches and confirmed by another mapper.
If I can recall, it has been like this for as long as I remember hitting 1000 brush entities last year or more. I didn't realize it was a problem until now.

**Edit: Hammer DOES compile these triggers into the BSP, as confirmed in a comment below. The problem lies in the source engine loading the 1000th brush-entity and beyond. Hammer is fine. The Source engine seems to be unable to load these brush-entities at runtime.**

### Steps to reproduce

1. - Create a test map with 1000 seperate triggers.
2. - Load map in-game.
3. - "sv_cheats 1"
4. - "showtriggers_toggle"
5. - The 1000th one (or any any other trigger after it) does not show up in-game with "showtriggers_toggle"

Hammer's compile log counts these missed entities.

### Image
https://cdn.discordapp.com/attachments/574831577373212677/736972294294863902/unknown.png
^Another mapper compiling 1100 trigger brushes. Hammer Log showed no errors and showed the correct full amount of brush entities in the editor, not 999. Only 999 show up in-game, the rest do not compile, chronologically after the 999th one.

**Comment by robotboy655** [comment]
Attach the .vmf file for testing?

**Comment by Armystuntman** [comment]
> Attach the .vmf file for testing?


[gm_hammer_limit_test_1011 vmf.zip](https://github.com/Facepunch/garrysmod-issues/files/4983338/gm_hammer_limit_test_1011.vmf.zip)
A map with 1010 triggers. Follow the above steps to see them.

**Comment by Cuffsss** [comment]
I did a bit of testing myself with Army in discord, after compiling the map, verifying the issue exists and then decompiling the map, we found that all 1010 trigger brushes were present in both the original vmf and the decompiled vmf. Hopefully, this points you in the right direction because I don't think the issue is related too the hammer editor nor the compilation process. The problem is likely an issue with Gmods build of the source engine.
